zcash_client_sqlite/src/wallet/init/migrations/shardtree_support.rs

```diff
@@ -101,19 +101,24 @@ impl RusqliteMigration for Migration {
             let mut block_rows = stmt_blocks.query([])?;
             while let Some(row) = block_rows.next()? {
                 let block_height: u32 = row.get(0)?;
-                let row_data: Vec<u8> = row.get(1)?;
+                let sapling_tree_data: Vec<u8> = row.get(1)?;
+                if sapling_tree_data == vec![0x00] {
+                    continue;
+                }
```

Comment:
Ideally I would have been able to use `NULL` values here, but we can't do that without rebuilding the table, and doing so would require breaking and re-forming all the foreign key constraints (which we can do with a pragma) but I concluded that using an invalid encoding and special-casing it would actually be the safest way to encode the "commitment tree removed/never present" state, because it will cause parse failures if we ever try to read it.

Reply:
Fixed now by updating the migration tests to store valid serialized note commitment trees.